### ISSUE TITLE
pull requires valid build path

### ISSUE THREAD

**Issue by tanmayambre** [docker/compose#3513]
```
docker-compose version
docker-compose version 1.7.0, build 0d7bf73
docker-py version: 1.8.0
CPython version: 2.7.9
OpenSSL version: OpenSSL 1.0.1e 11 Feb 2013

```

I have a docker-compose file 

```
version: '2'
services:
    xyzservice:
    image: xxx-image
    container_name: xxx_container
    build:
      context: ./somecontext
      dockerfile: Dockerfile.xxx
    networks:
      - some-network

```

When I want to build the image using `docker-compose build`, I understand that the **_Dockerfile.xxx**_ is absolutely required.

But if I just want to pull the image using `docker-compose pull` why is the **Dockerfile.xxx** required?

I get the following error on pull command

```
ERROR: build path ./somecontext either does not exist, is not accessible, or is not a valid URL.
```

**Comment by shin-** [comment]
You don't need the `build` section if you're just going to pull the image. Your compose file should be

```yaml
version: '2'
services:
    xyzservice:
    image: xxx-image
    container_name: xxx_container
    networks:
      - some-network

```

**Comment by tanmayambre** [comment]
Thanks,
My scenario is the developer requires the buildpath to build & push the container while if someone wants to just run the container without build just pulling mage using same compose file.

**Comment by fvanderbiest** [comment]
This issue is annoying for our [use case](https://github.com/georchestra/georchestra/blob/16.12/docs/docker.md) where the compose file is used both to build and run the app. Valid build paths should **not** be required to run `docker pull` IMO.

**Comment by jeserkin** [comment]
@shin- does what if I am getting it without `build` section?
My `docker-compose.yml` is as follows:
```yml
version: "3"
services:
  app:
	image: busybox
    networks:
      - backend
  db:
    #image: postgres
	image: busybox
    networks:
      - backend

networks:
  backend:
    # Use a custom driver which takes special options
    driver: bridge
```

**Comment by Elyytscha** [comment]
this is annoying, just saying

**Comment by zypA13510** [comment]
Are you sure this is `kind/question`? @docker/maintainer

As noted by @**fvanderbiest** and other people, there are valid use cases where pulling without building is required. Could you please consider this as an enhancement?

**Comment by yamac-kurtulus** [comment]
This is seriously annoying and makes no sense. Could you please implement this?

**Comment by nojvek** [comment]
Just hit this, this is really annoying indeed.

My usecase is I want to build from github actions and push to registry on PR commit. When merged to master a separate job ssh into a digital ocean $5 pod and does `docker-compose pull` and docker-compose restart to pick up the latest changes.


This feels like a serious limitation. Why would someone need to rebuild if they can just pull ? 🤷‍♂

**Comment by Elyytscha** [comment]
its not stale its still annoying

**Comment by MartinSunal** [comment]
it seems to be fixed here #7564

**Comment by Elyytscha** [comment]
yep its fixed with 1.27.0, thanks alot! could be closed now i think
